### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by kbrummert**
## What GitHub issue does this PR apply to?

Resolves #xxxx

## Checklist

- [ ] I claimed any associated issue(s) and they are not someone else's
- [ ] I have looked at documentation to ensure I made any revisions correctly
- [ ] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by geekygirlsarah** [comment]
This is a change not requested and not even filled out the PR template correctly. It seems to be one of several tiny README changes made all in the same day, which I'm guessing is to get a Hacktoberfest shirt. I'm closing and marking as spam.

**Comment by kbrummert** [comment]
No.  I just started on GitHub (check my record).  I am learning how to use it, and was following tutorials on how to get started by fixing minor errors in documentation (check a million videos on YT).

It is really awful of you to jump to conclusions and discourage newbies who are genuinely trying to help by accusing us based on no evidence at all.  I know nothing about the t-shirt you are referring to, and t-shirts are a dime a dozen.

I will be sure to refrain from helping you in the future, and will warn others of your ungrateful and defensive attitude.  I would not want them to waste their time on your projects, only to be discouraged.

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Website that will compare language features side by side.
 
 ## Why would you want this?
-Good question. If there's an aspect of a language you don't know, you can compare a languages you know with a language you do. It's a good way to quickly learn a new language, or use it as a quick reference to remember things by.
```
Comment (nightlark) [inline README.md:5]: Debates about the effort involved in this PR aside, this line as it is says you can compare a languages you know with a language you know; I imagine the intent was to say `compare a language you don't know with a language you do` or `compare languages you know with a language you don't`.

**Comment by geekygirlsarah** [comment]
Alright, @kbrummert, let's suppose I take you at your word.

First, [Hacktoberfest](https://hacktoberfest.digitalocean.com) is going on. It encourages developers to support open source projects by making 4 meaningful and quality pull requests against open source projects. The sponsoring companies will send swag to those that do it. I saw you have 5 repos , all of which have a tiny PRs against their README files, which go against the "quality pull requests". And due to the large amount of spam PRs they've had over the past few years, which look exactly like yours do, we are encouraged to mark them as invalid or spam. See the ["Values" section](https://hacktoberfest.digitalocean.com/resources) of their site, the ["Quality Standards" page](https://hacktoberfest.digitalocean.com/resources/qualitystandards), as well as the [resources for maintainers](https://hacktoberfest.digitalocean.com/resources/maintainers) section of Hacktoberfest's website, which is what I was following.

But assuming you actually don't know about Hacktoberfest and putting that aside, there is a template when you made the pull request you didn't fill out and explain what changed and why, or even checked the boxes confirming you took certain actions. There's documentation and contribution guidelines that I'm guessing you didn't read. In addition, I've had _dozens_ of new users to Github and new developers approach Code Thesaurus and have made much more quality PRs in the past month. (You're welcome to take a look at the nearly [200 past pull requests](https://github.com/codethesaurus/codethesaur.us/pulls?q=is%3Apr+is%3Aclosed+-label%3Aspam) on this project, of which [over 60](https://github.com/codethesaurus/codethesaur.us/pulls?q=is%3Apr+updated%3A%3E2021-09-30+is%3Aclosed) have been this month of October 2021 alone). So compared to those, this also looks like a bit of a spammy request.

So this is my dilemma, Kurt. Yours happens to be the only real tiny change during the spammiest open source month in which tiny README changes are exactly what the majority of spam people do. But @nightlark is also right, there probably is a different intent to the sentence.

I like to try to be encouraging and supportive to new developers, and I've been trying to do that for over a decade now. I'll go ahead and reopen this, but I may wait to merge it in until after the Hacktoberfest review period to do that.

**Comment by kbrummert** [comment]
Below is the video I watched.  One of the key points of the video--and many others like it--is that anyone can contribute to GitHub in small ways, even if it is only a few grammatical details in the documentation.  The assumption is that any help is appreciated, and that newbies need not be intimidated or expected to make complex code changes.  Based on this experience on my very first day of contributions in GitHub, that assumption appears to be wrong.

My understanding of the spirit of open-source--from everything I've watched and read--is that the size of the change is irrelevant.  Your statement that other contributors "have made much more quality PRs in the past month" is a judgement call on your part, and nothing more.

What you perceive as a dilemma is really a situation where you are fortunate to have been offered more help than you need, and now appear to be taking a lot of that help for granted.  Behind what you call "spammy" requests are real people with thoughts and feelings who may want to help and/or develop their skills.  It is not your place to judge their intentions without sufficient evidence to assume otherwise.  My hope is that they will not be discouraged, and will continue to be helpful to others.

I don't know if those Hacktoberfest t-shirts are gold-plated or made from cashmere, but even if they are, and I happen to win one for my "tiny, spammy" contributions, I hereby officially announce on GitHub that I give up my right to one, and ask that it be given to someone who needs a shirt.


https://www.youtube.com/watch?v=waEb2c9NDL8

**Comment by geekygirlsarah** [comment]
> 
> 
> ...is that anyone can contribute to GitHub in small ways, even if it is only a few grammatical details in the documentation. The assumption is that any help is appreciated...

Yes, this is generally the case. And yes, I've taken in PRs before of small, minor changes.

>... and that newbies need not be intimidated or expected to make complex code changes.

Yes. And I have written the contributing docs, as well as the documentation site, to reflect an openness and supportive type of place to make those changes, no matter one's experience level.

>Based on this experience on my very first day of contributions in GitHub, that assumption appears to be wrong.
> 
> My understanding of the spirit of open-source--from everything I've watched and read--is that the size of the change is irrelevant. Your statement that other contributors "have made much more quality PRs in the past month" is a judgement call on your part, and nothing more.

I did make this decision, but I didn't make it in jest. I looked at your profile. I looked at the repos you cloned. I saw the tiny changes. I also took note of the time of year it was, I took note in the content of the PRs you made. I also re-reviewed Hacktoberfest's documentation, and consulted with other open source maintainers I know. It was then after _**all**_ of that, I marked it spam and I wrote the comment mentioning why.

> What you perceive as a dilemma is really a situation where you are fortunate to have been offered more help than you need, and now appear to be taking a lot of that help for granted.

This is where you're wrong. I don't have enough help. The scope of work on CT is immense. It's why I made it open source. I knew I could never accomplish this on my own. And I am VERY fortunate to have had over 75 contributors who have made hundreds of contributions. (And go look over my comments. I thank and appreciate all of them.) This project has only grown at the scale it has and gotten the good attention it has by the wonderful people that have worked on it. People bring it up on Hacktoberfest's Discord and on Twitter as a great project to work on.

>Behind what you call "spammy" requests are real people with thoughts and feelings who may want to help and/or develop their skills. 

In the past decade, I've given conference talks, given workshops, taught university programming classes, and have mentored people from elementary school to adults. I do help people develop their skills and level up their careers. I understand these people as I've had to work with literally hundreds of people over the years grow and understand how they think and learn. So yes, I understand there are real people.

>It is not your place to judge their intentions...

But it _**IS**_ my place to judge these. You, and the 75+ other people, have made contributions to my project. You are literally _**requesting**_ me (and the other maintainers) to _**pull**_ in your code. "Pull request." I have to spend my own time to look over every bit of code, every bit of documentation, every bit of language updates, and more, and test and validate it all. If I pull in bad code or documentation without validating it, it brings down my project. At best, it becomes less usable. At worse, it's no longer usable. I don't do this because I'm bored. I'm building the tool I've needed for the majority of my development career.

If you are new to the development world, then congrats! I will gladly welcome you in. But you will _**never**_ land at a role or project in your career where your work won't go through similar scrutiny. On a good team, you'll be faced with empathetic people that may tell you your code doesn't work and offer suggestions. On a bad team, they'll tell you that you suck and your work is no good. (I honestly hope you never had to experience this as I have.) But either way, you can't just make effortless code changes, and you can't make changes without consequences. After decades of programming, I know what I'm doing, but I still approach my changes with caution because those changes can have long-lasting impacts.

>... without sufficient evidence to assume otherwise.

I had sufficient evidence. That's what this all comes down to. Your request looked spammy. My research into it backed it up. Resources tell maintainers to mark it as "spam". I did. I took action based on my data at the time, which was correct at the time, and also my responsibility as a leader on this project. That should hopefully validate that I think you're "real people with thoughts and feelings who may want to help and/or develop their skills". I hope also that you realize this project is larger than just you, is very established where it's at, that Hacktoberfest is a strong community, and whether your contribution was "good" or "bad", it gave off a spammy appearance.

> I don't know if those Hacktoberfest t-shirts are gold-plated or made from cashmere...

No, regular, run-of-the-mill cotton shirts. I've worked to earn my 4 Hacktoberfest shirts. But some don't. But in an industry that values entitled "brilliant" people and also has an over-valued sense that only the "best" people work on open source, and therefore people start to do whatever they can to make it look like they're smarter than they are or more active in open source than they are, the whole open source system is off, and has been even before Hacktoberfest. 

>https://www.youtube.com/watch?v=waEb2c9NDL8

I watched this. There's a couple of things I notice.

image

First, he takes the PR template and just deletes most of it. This is bad. He later says he is a maintainer on the project. So whatever he does as a PR to the project essentially means it doesn't matter. This is misleading on his part.

Second, he doesn't talk about contribution guidelines. I'd say a vast majority of projects that want contributions from people provide these guidelines, often in a CONTRIBUTING.md file in the root of the project. They talk about how to work on the project. If you don't follow the guidelines, you can probably expect your changes to get rejected. You didn't follow the instructions on the PR template when you made this PR, and even when GitHub said "it looks like you're a first time contributor. You should read over the guidelines.", it looks like you likely didn't, and I have every right to approve or reject any code into my project.

So in conclusion: If you continue to do open source work, I wish you the best of luck. If you decide to contribute to CT, then I'll welcome all meaningful changes, especially when it comes to [changes that are requested](https://github.com/codethesaurus/codethesaur.us/issues?q=is%3Aopen+is%3Aissue+no%3Aassignee). If you don't, then I hope you find other projects to work on that are meaningful to you. But either way, you should value the maintainer(s)'s time, but also be open to learning and growing. Almost all of your code will have to be reviewed to some extent as you go about your career, and you'll need to learn to take those updates in. You'll need to learn to follow directions too.

Regardless of whatever you do from here on out, I've now spent plenty of my time now writing a thorough and thoughtful response to you. And hopefully me spending a ton of time responding this to this proves I see you as "real people with thoughts and feelings who may want to help and/or develop their skills."

**Comment by kbrummert** [comment]
It doesn't prove anything, except that you will spend considerable time and effort rationalizing your behavior, when a simple thank-you for my efforts--however small--would have been sufficient and more appropriate.

All I did was send you a legit, constructive suggestion to fix a sentence that didn't appear to make sense, which the other commenter seemed to confirm.  Instead of just deciding to accept it or not--which would have been perfectly fine with me either way--you accused me of spamming and gaming the system to win a t-shirt I knew nothing about.

This is not about your level of experience, how many talks you've given, how many contributors you've had, how many newbies you've mentored, how many contributors you've thanked, how much time you've spent on your project, how hard you worked for your Hacktober t-shirts, how much time you've spent sifting through PRs, how many years you've been on GH, how much research you've done on my profile, etc....

All that matters in this case is that everyone should feel welcome on GitHub, regardless of tenure, experience level, number of contributions, or the depth of their contributions.  Again, that is the spirit of open-source.  No one should be judged, and no one should be met with unproven accusations, ever.

Normally, I avoid these types of drawn-out online debates, but my hope is that other newbies will read this and feel like they can stand up for themselves, and never feel they need to slink away from GH thinking they weren't cut out for programming, just because someone who has been here longer feels they are in a position to judge them.  No one should ever be made to feel that way.

**Comment by geekygirlsarah** [comment]
And I took the time (which I don't normally do) to tell you _**why**_ your PR came across as spammy, whether it was or not. You did _**not**_ follow the contribution guidelines and the instructions in the PR. That's on you. By that logic, you won't get far in your development career. And _**THAT'S**_ why I spent so long writing out longer answers for you. Because if you are a beginner, and you're trying to submit a valid PR, and even if I'm wrong and this isn't spam, then you should correct your expectations of what's happening. I think the project's track record of being great for beginners will speak for itself.

**Pull request closed without merging**